# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from statsmodels.stats.proportion import proportions_ztest

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# Number of sample size for black-sounding names and white-sounding names
b_total = len(data[data.race=='b'])
w_total = len(data[data.race=='w'])
print('Sample size for black-sounding names: {}'.format(b_total))
print('Sample size for white-sounding names: {}'.format(w_total))

Sample size for black-sounding names: 2435
Sample size for white-sounding names: 2435


In [4]:
# Number of callbacks for black-sounding names and white-sounding names
b_call = sum(data[data.race=='b'].call)
w_call = sum(data[data.race=='w'].call)

In [5]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [6]:
# Compute margin of error for 5% significance level
p = sum(data.call) / len(data)
sigma = np.sqrt((2*p*(1-p))/b_total)
margin_of_error = sigma*1.96
print('The margin of error for 5 percent significance level is: %.4f' % margin_of_error)

The margin of error for 5 percent significance level is: 0.0153


In [7]:
# Calculating the 95% confidence interval
upper = 0 + margin_of_error
lower = 0 - margin_of_error
print('The 95 percent confidence interval is: (%.4f, %.4f)' % (lower, upper))

The 95 percent confidence interval is: (-0.0153, 0.0153)


In [8]:
# Calculating the two samples proportions difference
p_b = b_call / b_total
p_w = w_call / w_total
p_sample_diff = p_b - p_w
print('The difference of the two samples callbacks proportions is: %.4f' % p_sample_diff)

The difference of the two samples callbacks proportions is: -0.0320


In [9]:
# Calculating the z statistic and p value
counts = np.array([b_call, w_call])
nobs = np.array([b_total, w_total])
zstat, pval = proportions_ztest(counts, nobs, alternative='two-sided')
print('The z statistic is: %1.2f and the p value is: %.5f' % (zstat, pval))

The z statistic is: -4.11 and the p value is: 0.00004


### Answers: 
1. Since the sample sizes are both larger than 30, the CLT applies and z test is appropriate to use. 

2. Null Hypothesis: There is no difference between black sounding names and white sounding names on the rate of callbacks for resumes. Alternate Hypothesis: There is difference between black sounding names and white sounding names on the rate of callbacks for resumes.

3. Margin of error is 0.0153 for 5% significance level and the 95% confidence interval is (-0.0153, 0.0153). The p value for hypothesis test is 0.00004. 

4. In this hypothesis, 5% significance level is suggested. Since the p value is much lower than 5%, the null hypothesis is rejected and with less than 5% probability of type I error, we can say that there is difference between black sounding names and white sounding names on the rate of callbacks for resumes. We can also see this from the fact that the difference of the two samples proportions is far away from the 95% confidence interval. 

5. My analysis does not mean that the race/name is the most important factor in callback success. In order to get a sense of this, I would need to do statistical analysis on the effect of other features to the callback proportions. 